File: 基礎題庫/Contents/a006/a006.py
import sys
import math

for num in sys.stdin:
    a,b,c = map(int,num.split())
    t = b**2 - 4*a*c
    if t < 0:
        print("No real root")
    elif t == 0:
        print("Two same roots x=" + str(int(-b/(2*a))))
    else:
        x1 = (-b + math.sqrt(t))/(2*a)
        x2 = (-b - math.sqrt(t))/(2*a)
        print("Two different roots x1=" + str(int(x1)) + " , x2=" + str(int(x2)))

File: 基礎題庫/Contents/a006/a006.cpp
#include<iostream>
#include<cmath>

using namespace std;

int main(void)
{
	int x1, x2;
	int a, b, c, d;
	
	while(cin >> a >> b >> c)
	{
		d = b * b - 4 * a * c;
		
		if(d < 0)
			cout << "No real root" << endl;
		else if(d == 0)
			cout << "Two same roots x=" <<  -b / (2 * a) << endl;
		else
		{
			x1 = (-b + sqrt(d)) / (2 * a);
			x2 = (-b - sqrt(d)) / (2 * a);
			cout << "Two different roots x1=" << x1 << " , x2=" << x2 << endl;
		}
	}
	
	return 0;
}
